htap.go

```diff
@@ -1,8 +1,28 @@
 package gosnowflake
 
+import "sync"
+
 type queryContextEntry struct {
 	ID        int   `json:"id"`
 	Timestamp int64 `json:"timestamp"`
 	Priority  int   `json:"priority"`
 	Context   any   `json:"context,omitempty"`
 }
+
+type queryContextCache struct {
+	mutex   *sync.Mutex
+	entries []queryContextEntry
```

Comment:
How to restrict the size of the cache? This value comes from server, if you don't see value then initialize with 5 Max entries in the cache.

Reply:
Yes, I have a task for it: https://snowflakecomputing.atlassian.net/browse/SNOW-895536
But I wanted to do it after the implementing prioritizing.